src/metabase/api/dataset.clj

```diff
@@ -51,29 +52,31 @@
       :or   {context       :ad-hoc
              export-format :api
              qp-runner     qp/process-query-and-save-with-max-results-constraints!}}]
```

Comment:
While not strictly part of the #33499 issue, `dataset` is a slow endpoint that needs instrumentation.